packages/firestore/src/remote/persistent_stream.ts

```diff
@@ -251,10 +251,10 @@ export abstract class PersistentStream<
   markIdle(): void {
     this.idle = true;
     this.queue
-      .schedule(() => {
+      .scheduleWithDelay(() => {
         return this.handleIdleCloseTimer();
       }, IDLE_TIMEOUT_MS)
-      .catch((err: FirestoreError) => {
+      .promise.catch((err: FirestoreError) => {
```

Comment:
Would it be possible to do something along the lines of https://github.com/alkemics/CancelablePromise/blob/master/CancelablePromise.js?

We might just want to expose .catch/.then, and keep .promise to make more advanced operations possible.